app/models/collection.js

```diff
@@ -28,12 +26,27 @@ class Collection {
             _.extend(this, props);
         }
     }
+
+    deserializeBullets(bulletInstances) {
+        this.bullets = this.bullets.map(bulletId => {
+            const bullet = bulletInstances.find(b => b.id === bulletId);
+            return new Bullet[bullet.type](bullet);
+        });
+        return this;
+    }
+
+    serializeBullets() {
+        if(this.bullets.every(b => typeof b !== "string")){
+            this.bullets = this.bullets.map(bullet => bullet.id); //beforeSave, converts bullet instances to ids
+        }
+    }
+
     addBullet(bullet, index) {
-        var index = index || this.bullets.length;   //so we can preserver ordering in collections.bullet array
-        this.bullets = this.bullets.slice(0, index).concat(bullet.id).concat(this.bullets.slice(index));
+        index = index || this.bullets.length; //so we can preserver ordering in collections.bullet array
+        this.bullets = this.bullets.slice(0, index).concat(bullet).concat(this.bullets.slice(index));
         if (bullet.collections.indexOf(this.id) < 0) bullet.collections.push(this.id)
         return Promise.all([this.save(), bullet.save()])
-        .catch(err => console.error('error ', err))
+            .catch(err => console.error('error ', err))
```

Comment:
This seems OK for now, but you might consider using something like [growl](https://github.com/marcorinck/angular-growl) for errors that are relevant to the end user.